### PR TITLE
Fix pretty printing remote name in pp_expr function

### PR DESCRIPTION
**Pull request by Premwoik**
This PR addresses #42 and adds missing pp for a remote name that is used by the `call_intersect` error formatter.

I also added a test case that confronts `pp_expr` with the sent example from the issue. The produced result is not fully correct and it needs further investigation.

I don't know why one fmt test won't work on CI when works locally:
```
code: res = ElixirFmt.format_error(error, [])
     stacktrace:
       :erlang.++("(non_error_kind(), payload, stacktrace() -> payload)", [])
       (gradualizer 0.0.0+build.1009.ref650998d) /home/runner/work/gradient/gradient/deps/gradualizer/src/gradualizer_fmt.erl:505: :gradualizer_fmt.pp_intersection_type/2
```

### CODE REVIEW

File: test/gradient/elixir_expr_test.exs [hunk 1]
```diff
@@ -326,4 +326,69 @@ defmodule Gradient.ElixirExprTest do
                actual
     end
   end
+
+  test "pp and format complex try expression" do
+    {_tokens, ast} =
+      Gradient.TestHelpers.load("/Elixir.CallRemoteException.beam", "/call_remote_exception.ex")
```
Comment (erszcz) [inline test/gradient/elixir_expr_test.exs:332]:
```suggestion
      Gradient.TestHelpers.load("Elixir.CallRemoteException.beam", "call_remote_exception.ex")
```

**Comment by erszcz** [comment]
> I don't know why one fmt test won't work on CI when works locally

Hmm, interesting, I'll take a closer look at it, but will leave the PR open in the meantime.

**Comment by erszcz** [comment]
@Premwoik I see the CI passes now. Is there anything else left to do in this PR?

**Review by erszcz — APPROVED** [review]

**Comment by Premwoik** [comment]
> @Premwoik I see the CI passes now. Is there anything else left to do in this PR?

I think that the main goal is completed, and we can merge.

**Pull request merged**